test/lib/custom-assertions.js

```diff
@@ -204,37 +208,24 @@ function assertSegments(parent, expected, options) {
  */
 function match(actual, expected) {
   if (typeof actual === 'string' && typeof expected === 'string') {
-    const patterns = expected
-      .trim()
-      .split(/\r?\n/)
-      .map((s) => s.trim())
-
-    let lastIndex = -1
-    for (const pattern of patterns) {
-      const index = actual.indexOf(pattern)
-      if (index === -1 || index < lastIndex) {
-        return false
-      }
-      lastIndex = index
-    }
-    return true
+    assert.ok(actual.indexOf(expected) > -1)
+    return
   }
 
   for (const key in expected) {
     if (key in actual) {
-      if (typeof expected[key] === 'object' && expected[key] !== null) {
-        /* c8 ignore next 3 */
-        if (!match(actual[key], expected[key])) {
-          return false
-        }
-      } else if (actual[key] !== expected[key]) {
-        return false
+      if (typeof expected[key] === 'function') {
+        const type = expected[key]
+        assert.ok(typeof actual[key] === typeMappings[type.name])
+      } else if (expected[key] instanceof RegExp) {
+        assert.ok(expected[key].test(actual[key]))
```

Comment:
I have a feeling that the documentation needs to be updated to illustrate these cases.